### PR TITLE
update: updated MaxSignatureAttempts logic

### PR DESCRIPTION
**Pull request by patrickzheng200**
This PR updates logic of MaxSignatureAttempts based on issue #212.

Signed-off-by: Patrick Zheng <patrickzheng@microsoft.com>

### CODE REVIEW

File: notation.go [hunk 1]
```diff
@@ -164,15 +164,14 @@ func Verify(ctx context.Context, verifier Verifier, repo registry.Repository, op
...
 	errExceededMaxVerificationLimit := ErrorVerificationFailed{Msg: fmt.Sprintf("total number of signatures associated with an artifact should be less than: %d", opts.MaxSignatureAttempts)}
 	numOfSignatureProcessed := 0
 	err = repo.ListSignatures(ctx, artifactDescriptor, func(signatureManifests []ocispec.Descriptor) error {
 		// process signatures
 		for _, sigManifestDesc := range signatureManifests {
-			if numOfSignatureProcessed >= opts.MaxSignatureAttempts {
+			// user specifies a positive opts.MaxSignatureAttempts
+			// Tracked by issue: https://github.com/notaryproject/notation-go/issues/212
+			if opts.MaxSignatureAttempts > 0 && numOfSignatureProcessed >= opts.MaxSignatureAttempts {
 				break
```
Comment (JeyJeyGao) [inline notation.go:175]: Can we return `errExceededMaxVerificationLimit` directly?

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Comment by priteshbandi** [comment]
Why are we updating this logic? Can we keep this code and caller can pass math.MaxInt64 which is unlimited for all practical purpose? These not only simplifies our code logic and makes it more readable for caller.

**Comment by patrickzheng200** [comment]
> Why are we updating this logic? Can we keep this code and caller can pass math.MaxInt64 which is unlimited for all practical purpose? These not only simplifies our code logic and makes it more readable for caller.

Just want to clarify, the original logic has the `opts.MaxSignatureAttempts <= 0` check, and the default of `opts.MaxSignatureAttempts` is 0. It means that caller has to pass a value to `opts.MaxSignatureAttempts`, otherwise notation.Verify() would fail. Is this what we want? This PR intends to remove this check, so if caller ignore this field, notation.Verify() will still process and verify unlimited number of signatures.

**Comment by shizhMSFT** [comment]
Let's close this PR due to security considerations on unlimited signature verification.

**Comment by patrickzheng200** [comment]
> Let's close this PR due to security considerations on unlimited signature verification.

Makes sence. PR closed.

**Comment by patrickzheng200** [comment]
> Why are we updating this logic? Can we keep this code and caller can pass math.MaxInt64 which is unlimited for all practical purpose? These not only simplifies our code logic and makes it more readable for caller.

Makes sence. Closed this PR, instead will set a large number such as math.MaxInt64 inside notation CLI.

**Pull request closed without merging**